tests/v2/validation/pipeline/hapostinstall/main.go

```diff
@@ -0,0 +1,182 @@
+
+	clusterID = "local"
+
+	configFileName       = "cattle-config.yaml"
+	environmentsFileName = "environments.groovy"
+
+	tokenEnvironmentKey      = "HA_TOKEN"
+	kubeconfigEnvironmentKey = "HA_KUBECONFIG"
+	configEnvironmentKey     = "CATTLE_TEST_CONFIG"
+)
+
+type wrappedConfig struct {
+	Configuration *rancher.Config `yaml:"rancher"`
+}
+
+func main() {
+	rancherConfig := new(rancher.Config)
+	rancherConfig.Host = host
+	isCleanupEnabled := false
+	rancherConfig.Cleanup = &isCleanupEnabled
+
+	adminUser := &management.User{
+		Username: "admin",
+		Password: adminPassword,
+	}
+
+	//create admin token
+	var adminToken *management.Token
+	err := kwait.Poll(500*time.Millisecond, 5*time.Minute, func() (done bool, err error) {
+		adminToken, err = token.GenerateUserToken(adminUser, host)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		logrus.Fatalf("error creating admin token", err)
+	}
+	rancherConfig.AdminToken = adminToken.Token
+
+	//create config file
+	configWrapped := &wrappedConfig{
+		Configuration: rancherConfig,
+	}
+	configData, err := yaml.Marshal(configWrapped)
+	if err != nil {
+		logrus.Fatalf("error marshaling", err)
+	}
+	err = os.WriteFile(configFileName, configData, 0644)
+	if err != nil {
+		logrus.Fatalf("error writing yaml", err)
+	}
+
+	//set config path
+	configPath, err := getConfigPath()
+	if err != nil {
+		logrus.Fatalf("error getting work path", err)
+	}
+	os.Setenv(configEnvironmentKey, configPath)
+	defer os.Unsetenv(configEnvironmentKey)
+
+	//generate kubeconfig
+	session := session.NewSession()
+	client, err := rancher.NewClient("", session)
+	if err != nil {
+		logrus.Fatalf("error creating client", err)
+	}
+
+	err = updateEULA(client)
+	if err != nil {
+		logrus.Fatalf("error updating EULA", err)
+	}
+
+	cluster, err := client.Management.Cluster.ByID(clusterID)
+	if err != nil {
+		logrus.Fatalf("error getting cluster", err)
+	}
+	kubeconfig, err := client.Management.Cluster.ActionGenerateKubeconfig(cluster)
+	if err != nil {
+		logrus.Fatalf("error getting kubeconfig", err)
+	}
+
+	//create groovy environments file
+	kubeconfigb64 := b64.StdEncoding.EncodeToString([]byte(kubeconfig.Config))
+	kubeconfigEnvironment := generateEnvStr(kubeconfigEnvironmentKey, kubeconfigb64)
+	tokenEnvironment := generateEnvStr(tokenEnvironmentKey, adminToken.Token)
+	environmentsData := strings.Join([]string{tokenEnvironment, kubeconfigEnvironment}, "\n")
+	err = os.WriteFile(environmentsFileName, []byte(environmentsData), 0644)
+	if err != nil {
+		logrus.Fatalf("error writing yaml", err)
+	}
+
+}
+
+func generateEnvStr(key, value string) string {
+	prefix := "env"
+	return fmt.Sprintf("%v.%v='%v'", prefix, key, value)
+}
+
+func getConfigPath() (string, error) {
+	wd, err := os.Getwd()
+	fmt.Println("wd:", wd)
+	if err != nil {
+		return "", err
+	}
+
+	path := fmt.Sprintf("%v/%v", wd, configFileName)
+
+	return path, nil
+}
+
+func updateEULA(adminClient *rancher.Client) error {
```

Comment:
Should updateEULA be an extension? I could see other tests potentially using it.

Reply:
Moved this to `pipelineutil` package